src/Files/File.class.st

```diff
@@ -44,15 +46,218 @@ Class {
+{ #category : #'primitives-path' }
+File class >> deleteDirectory: fullPath [
+	"Delete the directory named by the given path. 
+	Fail if the path is bad or if a directory by that name does not exist."
+
+ 	<primitive: 'primitiveDirectoryDelete' module: 'FilePlugin'>
+	self primitiveFailed
+
+]
+
+{ #category : #'primitives-path' }
+File class >> deleteFile: aFileName [
+	"Delete the file of the given name. 
+	Return self if the primitive succeeds, nil otherwise."
+
+	<primitive: 'primitiveFileDelete' module: 'FilePlugin'>
+	^ nil
+
+]
+
+{ #category : #'primitives-path' }
+File class >> delimiter [
+	"Return the path delimiter for the underlying platform's file system."
+
+ 	<primitive: 'primitiveDirectoryDelimitor' module: 'FilePlugin'>
+	self primitiveFailed
+
+]
+
+{ #category : #'primitives-encoding' }
+File class >> encode: aString [
+
+	"Should not return a string"
+	^ ((ZnCharacterEncoder newForEncoding: 'utf8') encodeString: aString) asString
+]
+
+{ #category : #'primitives-file' }
+File class >> fileDescriptorType: fdNum [
+	"Allow to test if the standard input/output files are from a console or not
+	Return values:
+	* -1 - Error
+	* 0 - no console (windows only)
+	* 1 - normal terminal (unix terminal / windows console)
+	* 2 - pipe
+	* 3 - file
+	* 4 - cygwin terminal (windows only)"
+
+	<primitive: 'primitiveFileDescriptorType' module: 'FilePlugin' error: error>
+	error = #'bad argument'
+		ifTrue: [ fdNum isInteger
+				ifFalse: [ ^ self error: 'File Descriptor must be an integer!' ] ].
+	^ self primitiveFailed
+]
+
+{ #category : #'primitives-file' }
+File class >> flush: id [
+	"On Unix, the FilePlugin uses stdio FILE* structs which maintain their
+	own internal buffer to minimize write() syscalls. This flushes that buffer.
+	On Windows this and primSync: do the same thing."
+
+	<primitive: 'primitiveFileFlush' module: 'FilePlugin'>
+	
+	"We can't ignore fflush() failing, because it can fail for any of the
+	reasons write() can."
+	self primitiveFailed
+]
+
+{ #category : #'primitives-path' }
+File class >> getMacFile: fileName type: typeString creator: creatorString [
+	"Get the Macintosh file type and creator info for the file with the given name. Fails if the file does not exist or if the type and creator type arguments are not strings of length 4. This primitive is Mac specific; it is a noop on other platforms."
+
+ 	<primitive: 'primitiveDirectoryGetMacTypeAndCreator' module: 'FilePlugin'>
+
+
+]
+
+{ #category : #'primitives-file' }
+File class >> getPosition: id [
+	"Get this files current position."
+
+	<primitive: 'primitiveFileGetPosition' module: 'FilePlugin'>
+	self primitiveFailed
+
+]
+
 { #category : #'initialization-release' }
-File class >> initialize [
-	self useFilePlugin
+File class >> imageFile [
+	self 
+		deprecated: 'Use Smalltalk imageFile fullName instead'
+		on: 	'14 May 2018' 
+		in: 'Pharo-7.0+alpha.build.876.sha.f8079b5841caa282c5849479d7f16321aad0aead (32 Bit)'.
+	^ Smalltalk imageFile fullName
+]
```

Comment:
Why deprecating it? It was never a public method, and even less a method of this class. Nobody will call it. We can just remove it.